Doc/deprecations/pending-removal-in-3.17.rst

```diff
@@ -1,6 +1,17 @@
 Pending removal in Python 3.17
 ------------------------------
 
+* :mod:`hashlib`:
+
+  - Supporting the undocumented ``string`` keyword parameter in :func:`~hashlib.new`
+    and hash function constructors such as :func:`~hashlib.md5` is deprecated.
+    Use the ``data`` keyword parameter instead, or simply pass the data to hash
```

Comment:
For everywhere this statement is made I think we could word this better.  Let's not emphasize "undocumented" but instead just state what is happening and describe the past pre-3.13 API wart that people have encountered.  maybe something like this?

---

In hash function constructors such as :func:`hashlib.new` or the direct hash-named constructors such as :func:`hashlib.md5` and :func:`hashlib.sha256`, their optional initial data parameter could also be passed a keyword argument named ``data=`` and or  ``string=`` in various  :mod:`hashlib` implementations.  Support for the ``string=`` keyword argument name is now deprecated.  Prefer passing the initial data as a positional argument for maximum backwards compatibility.

Before Python 3.13, ...

---

Reply:
Thanks for the formulation! I'll use it. I intentionally didn't mention which constructors were affected because it depends on the Python version as well (I think something changed when we used HACL* BLAKE-2).

Reply:
yeah I don't want us to try and enumerate _which_ constructors did what when (your reply on the issue an actual understanding of this illustrates why that is madness :grin:).  I just like to give more than one example to get the point across that it was on "that type of function".

Reply:
> your reply on the issue an actual understanding of this illustrates why that is madness 

Look at the number of edits as well... it illustrates how confusing it was...